EXT: Resource Graph (#1746)
kyma-project/busola: docs/extensibility/resources.md

## _details_ section

> The **details** section defines the display structure for the details page. It contains three sections, `header`, `body` and optional `resourceGraph`. First two sections are a list of items to display in the **header** section and the body of the page respectively. The format of the entries is similar to the **form** section, however it has extra options available. The `resourceGraph` section is used to configure the ResourceGraph which shows relationships between various resources.

Reviewer: ```suggestion
The **details** section defines the display structure for the details page. It contains three sections, `header`, `body`, and optional `resourceGraph`. The first two sections are a list of items to display in the **header** section and the body of the page respectively. The format of the entries is similar to the **form** section, however it has extra options available. The `resourceGraph` section is used to configure the ResourceGraph which shows relationships between various resources.
```

}
```

### resourceGraph parameters

> - **depth** - defines the maximum distance from original resource to transitively related resource. Defaults to infinity.

Reviewer: ```suggestion
- **depth** - defines the maximum distance from the original resource to a transitively related resource. Defaults to infinity.
```

> - **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of node's border. If not set, the node's border is the same as current text color.

Reviewer: ```suggestion
- **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of node's border. If not set, the node's border is the same as the current text color.
```

> - **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of node's border. If not set, the node's border is the same as the current text color.

Reviewer: ```suggestion
- **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of the node's border. If not set, the node's border is the same as the current text color.
```

> - **networkFlowKind** - optional boolean which determines if the resource should be shown on network graph, Defaults to `false`, which displays the resource on structural graph.

Reviewer: ```suggestion
- **networkFlowKind** - optional boolean which determines if the resource should be shown on the network graph, Defaults to `false`, which displays the resource on the structural graph.
```

> - **networkFlowLevel** - optional integer which sets the horizontal position of resource's node on a network graph.

Reviewer: ```suggestion
- **networkFlowLevel** - optional integer which sets the horizontal position of the resource's node on the network graph.
```

- **dataSources** - an array of objects in shape:

>   - **source** - a string that must correspond to one of [dataSources](#datasources-section) name. It selects the related resource and the way it should be matched.

Reviewer: ```suggestion
  - **source** - a string that must correspond to one of the [dataSources](#datasources-section) name. It selects the related resource and the way it should be matched.
```

  - **namespace** - the resource's Namespace name; it defaults to the original resource's Namespace. If set to `null`, cluster-wide resources or resources in all Namespaces are matched.
  - **name** - a specific resource name; leave empty to match all resources of a given type.
- **ownerLabelSelectorPath** - the path to original object's **selector** type property; for example, `spec.selector.matchLabels` for Deployment, used to select matching Pods.

> - **filter** - [JSONata](https://docs.jsonata.org/overview.html) function enabling the user to write a custom matching logic. It can use the following variables:

Reviewer: ```suggestion
- **filter** - [JSONata](https://docs.jsonata.org/overview.html) function enabling the user to write a custom matching logic. It uses the following variables:
```

>   - **item** - the current item of a related kind.

Reviewer: ```suggestion
  - **item** - the current item of the related kind.
```
